scripts/PalettePanel.gd

```diff
@@ -1,36 +1,34 @@
 extends PanelContainer
 
+var foregroundPicker: ColorPickerButton
+
 # Called when the node enters the scene tree for the first time.
 func _ready():
-	pass # Replace with function body.
-
-# Called every frame. 'delta' is the elapsed time since the previous frame.
-#func _process(delta):
-#	pass
+	foregroundPicker = find_parent("Main").find_node("Foreground")
 
+# Adds the current foreground color to the palette
 func _on_AddColor_pressed():
-	var image = Image.new()
-	var texture = ImageTexture.new()
+	var image := Image.new()
+	var texture := ImageTexture.new()
 	
 	image.create(16, 16, false, Image.FORMAT_RGBA8)
-	
-	image.fill($Palette/Buttons/ColorPickerButton.color)
+	image.fill(foregroundPicker.color)
 	texture.create_from_image(image)
-	
 	$Palette/Colors.add_icon_item(texture)
```

Comment:
Consider creating a var colors for `$Palette/Colors` as it is used throughout this file.